Explorer/Assets/DCL/WebRequests/WebRequestController.cs

```diff
@@ -14,22 +18,81 @@ public class WebRequestController : IWebRequestController
         private readonly IWebRequestsAnalyticsContainer analyticsContainer;
         private readonly IWeb3IdentityCache web3IdentityCache;
         private readonly IRequestHub requestHub;
+        private readonly string csvPath;
+        private static readonly object csvLock = new object();
 
-        public WebRequestController(IWebRequestsAnalyticsContainer analyticsContainer, IWeb3IdentityCache web3IdentityCache, IRequestHub requestHub)
+        public WebRequestController(
+            IWebRequestsAnalyticsContainer analyticsContainer,
+            IWeb3IdentityCache web3IdentityCache,
+            IRequestHub requestHub)
         {
             this.analyticsContainer = analyticsContainer;
             this.web3IdentityCache = web3IdentityCache;
             this.requestHub = requestHub;
+
+            // Set up CSV file path in persistent data path
+            csvPath = Path.Combine(Application.persistentDataPath, "request_metrics.csv");
+
+            // Create CSV header if file doesn't exist
+            if (!File.Exists(csvPath))
+            {
+                lock (csvLock)
+                {
+                    if (!File.Exists(csvPath))
+                    {
+                        File.WriteAllText(csvPath,
+                            "Timestamp,RequestType,URL,RequestTime,OperationTime,TotalTime,Status,AttemptNumber,ErrorMessage\n");
+                    }
+                }
+            }
         }
 
-        public async UniTask<TResult?> SendAsync<TWebRequest, TWebRequestArgs, TWebRequestOp, TResult>(RequestEnvelope<TWebRequest, TWebRequestArgs> envelope, TWebRequestOp op)
+        private void LogMetricsToCSV(
+            string requestType,
+            string url,
+            long requestTime,
+            long operationTime,
+            string status,
+            int attemptNumber,
+            string errorMessage = "")
+        {
+            var timestamp = DateTime.UtcNow.ToString("yyyy-MM-dd HH:mm:ss.fff");
+            var totalTime = requestTime + operationTime;
+
+            // Sanitize values for CSV
+            url = url.Replace(",", "%2C");
+            errorMessage = errorMessage.Replace(",", ";").Replace("\n", " ").Replace("\r", "");
+
+            var logLine = $"{timestamp},{requestType},{url},{requestTime},{operationTime},{totalTime},{status},{attemptNumber},{errorMessage}\n";
+
+            lock (csvLock)
+            {
+                try
+                {
+                    File.AppendAllText(csvPath, logLine, Encoding.UTF8);
+                }
+                catch (Exception ex)
+                {
+                    ReportHub.LogError(
+                        null,
+                        $"Failed to write to metrics CSV: {ex.Message}"
+                    );
+                }
+            }
+        }

```

Comment:
This should happen on a background thread. Metrics messages go into a queue, and a job writes them to disk regularly. You don't want to be influencing the thing you're trying to measure. And again, the main thread is scarce resource.